src/app/views/apps-management/apps-management.controller.coffee

```diff
@@ -1,29 +1,65 @@
 angular.module 'mnoEnterpriseAngular'
   .controller('AppsManagementCtrl',
-    ($q, $scope, MnoeConfig, MnoeProductInstances, MnoeProvisioning, MnoeOrganizations, AppManagementHelper) ->
+    ($q, $scope, MnoeConfig, MnoeProductInstances, MnoeProvisioning, MnoeOrganizations, AppManagementHelper, MnoeAppInstances) ->
 
       vm = @
       vm.isLoading = true
       vm.recentSubscription = AppManagementHelper.recentSubscription
 
+      getSyncStatusValue = (product) ->
+        sync_status = _.find(vm.connec_apps, (app) -> app.uid == product.uid)
+        return 'Disconnected' unless sync_status?.status
+
+        if sync_status.status.toLowerCase() in ['error', 'disconnected']
+          'Disconnected'
+        else
+          'Connected'
+
+      getSyncStatuses = ->
+        vm.products = _.map(vm.products,
+          (product) ->
+            if product.uid in vm.filterSyncProductIds
+              product.sync_status = {}
+              product.sync_status.attributes = {}
+              product.sync_status.attributes.status = getSyncStatusValue(product)
+            product
+        )
+
       vm.providesStatus = (product) ->
         vm.dataSharingEnabled(product) || product.subscription
 
       vm.dataSharingEnabled = (product) ->
         MnoeConfig.isDataSharingEnabled() && product.data_sharing
 
       vm.dataSharingStatus = (product) ->
-        if product.sync_status?.attributes?.status
-          'Connected'
+        # If sync status is set from backend then it follows following:
+        # sync_status values:
+        #   null => disconnected
+        #   all other values => connected
+        if _.find(vm.filterSyncProductIds, (uid) -> uid == product.uid)
+          product.sync_status?.attributes?.status
         else
-          'Disconnected'
+          if product.sync_status?.attributes?.status
+            'Connected'
+          else
+            'Disconnected'
 
       vm.subscriptionStatus = (product) ->
         return product.subscription.status if product.subscription
 
       vm.appActionUrl = (product) ->
         "/mnoe/launch/#{product.uid}"
 
+      vm.initAppInstanceSync = ->
+        vm.filterSyncProductIds = _.map(_.filter(vm.products, (product) -> !product.sync_status), (prod) -> prod.uid)
```

Comment:
I'm thinking we should rename this to something like `vm.unsyncedProductIds` so it is clear that this is an array of product ids that do not have a sync status attached (until we fetch it from the 'app_instances_sync' endpoint)

Reply:
Also don't use lodash `map` and `filter` but [coffescript idioms](https://arcturo.github.io/library/coffeescript/04_idioms.html):
```coffeescript
vm.unsyncedProductIds = (prod.uid for prod in vm.products when !prod.sync_status)
```
Better perf (as it compiles to native JS  loop) and much easier to read!